### PR TITLE
Frontend plasma transfers

### PR DESCRIPTION
**Pull request by nicholasmueller**
Issue/Task: [990](https://github.com/omisego/ewallet/issues/990)

# Overview

Send transaction on plasma from internal wallet to any external address
Deposit to plasma from a cold wallet

# Usage

run `elixir-omg` and set `OMISEGO_CONTRACT_ADDRESS` in `eth_omisego_adapter` in elixir and `plasmaContractAddress` in frontend `/src/omg-plasma/config.js`

**For internal => external on plasma**
- make sure you have plasma balance in hot wallet
- make sure an internal primary wallet has a blockchain deposit address with a balance
- wallets page -> primary wallet -> external transfer button -> select token and option to transfer on plasma 
- you can see it working by sending to a registered cold wallet and the plasma balance should increase (blockchain wallets -> cold wallet detail)

**For cold wallet => deposit to plasma**
- make sure cold wallet has balance
- transfer -> select "to OmiseGO Network" -> confirm in metamask -> wait
- omg-js doesnt support callbacks during the lifecycle of the transaction so i shortcut the usual transfer flow (with pending and confirmed states)

### CODE REVIEW

File: apps/frontend/assets/src/omg-alert/reducer.js [hunk 1]
```diff
@@ -270,6 +270,12 @@ export const alertsReducer = createReducer([], {
       createAlertState('Submitted transaction to Ethereum successfully.', 'success')
     ]
   },
+  'WEB3/DEPOSIT/SUCCESS': state => {
```
Comment (unnawut) [inline apps/frontend/assets/src/omg-alert/reducer.js]: Should it be `WEB3/DEPOSIT/SUCCESS` if there's still the metamask approval step needed? Other success cases seem to be the completely successful one. Maybe `WEB3/DEPOSIT/ASK_APPROVAL`? Not sure.

File: apps/frontend/assets/src/omg-create-blockchain-transaction-modal/index.js [hunk 2]
```diff
@@ -99,9 +99,14 @@ class CreateBlockchainTransaction extends Component {
 
   setGasPrice = async () => {
     const { estimateGasFromTransaction } = this.props
+    if (this.state.toPlasma) {
+      return this.setState({ gasPrice: 500 })
```
Comment (unnawut) [inline apps/frontend/assets/src/omg-create-blockchain-transaction-modal/index.js]: Should we move the price out in to a common class?

Also why 500?

File: apps/frontend/assets/src/omg-create-blockchain-transaction-modal/index.js [hunk 3]
```diff
@@ -333,6 +340,14 @@ class CreateBlockchainTransaction extends Component {
         <AllBlockchainWalletsFetcher
           render={({ blockchainWallets }) => {
             const hotWallets = blockchainWallets.filter(i => i.type === 'hot')
+            const wallets = [
+              ...hotWallets,
+              {
+                address: 'OmiseGO Network',
```
Comment (unnawut) [inline apps/frontend/assets/src/omg-create-blockchain-transaction-modal/index.js]: Address is 'OmiseGO Network'`?

File: apps/frontend/assets/src/omg-internal-to-external-modal/index.js [hunk 4]
```diff
@@ -245,13 +256,79 @@ class CreateTransaction extends Component {
...
+                query={{ address }}
+                render={({ data, individualLoadingStatus }) => {
+                  if (individualLoadingStatus !== 'SUCCESS') {
+                    return this.renderPlasmaLoadingState()
+                  }
+                  const tokenBalances = _.find(data, i => i.token.id === this.state.fromTokenSelected.token.id)
+                  const { plasmaAmount } = tokenBalances
+
+                  const internalAmount = this.state.fromTokenSelected.amount
+                  const onPlasma = plasmaAmount > internalAmount
```
Comment (unnawut) [inline apps/frontend/assets/src/omg-internal-to-external-modal/index.js:296]: :mindblown:

File: apps/frontend/assets/src/omg-transaction/CreateBlockchainTransactionButton.js [hunk 5]
```diff
@@ -15,7 +15,7 @@ function CreateBlockchainTransactionButton (props) {
       }
       disabled={!window.web3 || !window.ethereum}
     >
-      <span>Topup</span>
+      <span>Transfer</span>
```
Comment (unnawut) [inline apps/frontend/assets/src/omg-transaction/CreateBlockchainTransactionButton.js:18]: 👍

**Comment by unnawut** [comment]
Note that this is purely code review. Didn't try run.

File: apps/frontend/assets/src/omg-alert/reducer.js (hunk 1 above)
Comment (nicholasmueller) [reply apps/frontend/assets/src/omg-alert/reducer.js]: ah thats a good point. will update to make this more accurate 👍

File: apps/frontend/assets/src/omg-create-blockchain-transaction-modal/index.js (hunk 2 above)
Comment (nicholasmueller) [reply apps/frontend/assets/src/omg-create-blockchain-transaction-modal/index.js]: @unnawut unlike the eth transfer I don't know how to estimate the cost of transfer for a contract call so the 500 is a pretty random default. @jarindr maybe has an idea to estimate contract call gas? I don't know what amount would be a better default?

File: apps/frontend/assets/src/omg-plasma/config.js [hunk 6]
```diff
@@ -3,7 +3,8 @@ export default {
   web3ProviderUrl: 'http://localhost:8545',
   watcherUrl: 'http://localhost:7434',
   childchainUrl: 'http://localhost:9656',
-  plasmaContractAddress: '0xb0df2ea778e5ddc6c85d25fbed335e67211a15b5',
+  plasmaContractAddress: '0xc887b342280bd175bb66e1d9323df0d52a83611a',
```
Comment (T-Dnzt) [inline apps/frontend/assets/src/omg-plasma/config.js:6]: Shouldn't we get this from some config somehow?

Comment (jarindr) [reply apps/frontend/assets/src/omg-plasma/config.js:6]: this should be replaced with the blockchain config that is setup on backend @T-Dnzt

Comment (nicholasmueller) [reply apps/frontend/assets/src/omg-plasma/config.js:6]: @T-Dnzt can i merge this and make new pr implementing O's blockchain settings?

**Pull request closed without merging**